validation/go.mod

```diff
@@ -5,8 +5,10 @@ go 1.21
 replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain
 
 require (
+	github.com/ethereum-optimism/optimism v1.4.2
```

Comment:
Dependency on monorepo here is fine, since this module doesn't get pulled into the monorepo. Still a bit odd how the `go.mod` was outdated. Is the `validation` module not checked in Go CI?

Reply:
Apparently not, I really just saved the file and my dev env did the rest here. I then ran `go mod tidy` in both modules to make sure everything's up to date.